### PR TITLE
Fix missing bio_new

### LINKED ISSUE #2: missing bio_new
```
Error in line 1: while requiring "./spec/x509_spec.cr"

in spec/x509_spec.cr:30: instantiating 'OpenSSL::X509::Certificate.class#new(String)'

    from_pem = Certificate.new pem
                           ^~~

in src/openssl_ext/x509/certificate.cr:7: instantiating 'OpenSSL::GETS_BIO.class#new(IO::Memory)'

      bio = OpenSSL::GETS_BIO.new(io)
                              ^~~

in src/openssl_ext/bio.cr:27: undefined fun 'bio_new' for LibCrypto

    @bio = LibCrypto.bio_new(pointerof(GETS_BIO))

```

### PR DESCRIPTION
**Pull request by noellabo**
- Resolve errors related to 'bio_new' occurring in crystal 0.27.0

Fixes #2

### CODE REVIEW

**Comment by byfoz** [comment]
@noellabo huge thanks for this! 🙌🏻

Just want to check before merging in:
> I was worried about errors, so I wrote a fix.
>
> It will work with OpenSSL 1.0.2, but because it is not solving the essential problem in 1.1 and later it should not work yet.

Can you elaborate on this? ...
>but because it is not solving the essential problem in 1.1 and later it should not work yet

I'm not sure what the essential problem is in this context - it would be great if we can support more than OpenSSL v1.0.2

**Comment by noellabo** [comment]
Fixed the problem that the spec did not pass.

- Since the key length of RSA specified in spec is too short in OpenSSL 1.1.1, we changed the minimum value to 512. (Segmentation fault and some failures occurred.)
- In PEM decode including blank lines, returning 0 to BIO_gets in OpenSSL 1.1.1 seems to be judged as a termination. As a workground, when reading only 1 byte, 1 is returned.

I think that the judgment of the blank line of gets of OpenSSL 1.1.1 is around here.
https://github.com/openssl/openssl/commit/98e553d2ce31e2179be68d6a60b5bec765cd9768#diff-77e13ca6978d539e6df17cdb50830082L387

If someone can make a better correction, please solve it.

File: src/openssl_ext/bio.cr [hunk 1]
```diff
@@ -16,21 +16,28 @@ struct OpenSSL::GETS_BIO
...
     crystal_bio
   end
 
   @boxed_io : Void*
 
   def initialize(@io : IO)
-    @bio = LibCrypto.bio_new(pointerof(GETS_BIO))
+    @bio = LibCrypto.BIO_new(GETS_BIO)
 
     # We need to store a reference to the box because it's
```
Comment (CImrie) [inline src/openssl_ext/bio.cr]:
```suggestion

```

File: src/openssl_ext/bio.cr [hunk 2]
```diff
@@ -16,21 +16,28 @@ struct OpenSSL::GETS_BIO
...
     # stored in `@bio.value.ptr`, but that lives in C-land,
```
Comment (CImrie) [inline src/openssl_ext/bio.cr]:
```suggestion

```

File: src/openssl_ext/bio.cr [hunk 3]
```diff
@@ -16,21 +16,28 @@ struct OpenSSL::GETS_BIO
...
     # not in Crystal-land.
```
Comment (CImrie) [inline src/openssl_ext/bio.cr]:
```suggestion

```

**Review by byfoz — APPROVED** [review]
---

**Pull request merged**